hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMDirectoriesPurgeRequestWithFSO.java

```diff
@@ -255,7 +255,7 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager, Execut
           volBucketInfoMap.entrySet()) {
         entry.setValue(entry.getValue().copyObject());
       }
-      omMetadataManager.getLock().releaseWriteLocks(BUCKET_LOCK, bucketLockKeys);
+      mergeOmLockDetails(omMetadataManager.getLock().releaseWriteLocks(BUCKET_LOCK, bucketLockKeys));
```

Comment:
I think we should  only call `releaseWriteLocks`, if `isLockAcquired` is set to true (similar situation in `OmKeyPurgeRequest`). @swamirishi please help to advise here